airbyte-integrations/connectors/source-tiktok-marketing/source_tiktok_marketing/streams.py

```diff
@@ -204,10 +204,14 @@ def should_retry(self, response: requests.Response) -> bool:
         Once the rate limit is met, the server returns "code": 40100
         Docs: https://business-api.tiktok.com/marketing_api/docs?id=1701890997610497
         Retry 50002 as well - it's a server error.
+        Retry when 504 error: response doesn't consist json, so we need to handle response status code to retry.
         """
         try:
             data = response.json()
         except Exception:
+            if response.status_code == 504:
```

Comment:
I have two questions: 
* Do we expect Tiktok to maybe someday have the response body as JSON as well? If so, this code will not work anymore
* Why don't we add this as part of the HttpStream? I seems like whenever there will be a 504, this will be an error that we will want to have clear logging on. @brianjlai is there a reason why we would not put this in the CDK itself?

Reply:
Answering for first question, [docs](https://ads.tiktok.com/marketing_api/docs?id=1737172488964097), tik tok has declaretaled status codes in json response, there aren't any status for 504, which is html.